From 95a4737eae58b7f621c69eac48861a22d477fc90 Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@gmail.com>
Date: Wed, 24 Oct 2012 13:48:30 -0700
Subject: [PATCH 36/48] mutex: rb_init_node no more exists, remove it

No need to call rb_init_node, so remove it.
---
 kernel/futex.c   |    4 ++--
 kernel/rtmutex.c |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/kernel/futex.c b/kernel/futex.c
index f5b04b5..283d874 100644
--- a/kernel/futex.c
+++ b/kernel/futex.c
@@ -2297,8 +2297,8 @@ static int futex_wait_requeue_pi(u32 __user *uaddr, unsigned int flags,
 	 */
 	debug_rt_mutex_init_waiter(&rt_waiter);
 	rt_waiter.task = NULL;
-	rb_init_node(&rt_waiter.tree_entry);
-	rb_init_node(&rt_waiter.pi_tree_entry);
+	//rb_init_node(&rt_waiter.tree_entry);
+	//rb_init_node(&rt_waiter.pi_tree_entry);
 
 	ret = get_futex_key(uaddr2, flags & FLAGS_SHARED, &key2, VERIFY_WRITE);
 	if (unlikely(ret != 0))
diff --git a/kernel/rtmutex.c b/kernel/rtmutex.c
index 2f76bcc..f6a9074 100644
--- a/kernel/rtmutex.c
+++ b/kernel/rtmutex.c
@@ -720,8 +720,8 @@ rt_mutex_slowlock(struct rt_mutex *lock, int state,
 	int ret = 0;
 
 	debug_rt_mutex_init_waiter(&waiter);
-	rb_init_node(&waiter.tree_entry);
-	rb_init_node(&waiter.pi_tree_entry);
+	//rb_init_node(&waiter.tree_entry);
+	//rb_init_node(&waiter.pi_tree_entry);
 
 	raw_spin_lock(&lock->wait_lock);
 
-- 
1.7.9.5

